src/main/options.js

```diff
@@ -53,14 +53,30 @@ function normalize(options, opts) {
   rawOptions.astFormat = parser.astFormat;
   rawOptions.locEnd = parser.locEnd;
   rawOptions.locStart = parser.locStart;
-  rawOptions.printer = getPrinter(rawOptions);
+  const plugin = getPlugin(rawOptions);
+  rawOptions.printer = plugin.printers[rawOptions.astFormat];
+
+  const pluginDefaults = supportOptions
+    .filter(
+      optionInfo =>
+        optionInfo.pluginDefaults && optionInfo.pluginDefaults[plugin.name]
+    )
+    .reduce(
+      (reduced, optionInfo) =>
+        Object.assign(reduced, {
+          [optionInfo.name]: optionInfo.pluginDefaults[plugin.name]
+        }),
+      {}
+    );
 
   Object.keys(defaults).forEach(k => {
     if (rawOptions[k] == null) {
       rawOptions[k] = defaults[k];
     }
   });
 
+  Object.assign(rawOptions, pluginDefaults);
```

Comment:
Just realized that `rawOptions` already contains the default options, so assigning the "mixedDefaults" only if the option key is not present in rawOptions (line 73) won't work. I added some tests for it.

Reply:
Hmm, if I'm not mistaken the order of option application should be
```
core defaults < plugin defaults < cli / file overrides
```
That would mean that we already have to find out which plugin is used before CLI options are applied, so the logic has to be moved to somewhere around here:

 https://github.com/prettier/prettier/blob/master/src/cli/util.js#L224